block/internal/cache/generic_cache.go

```diff
@@ -178,86 +185,105 @@ func (c *Cache[T]) SaveToDisk(folderPath string) error {
-		keyStr, okKey := k.(string)
-		valUint64, okVal := v.(uint64)
-		if okKey && okVal {
-			daIncludedToSave[keyStr] = valUint64
+	wg.Go(func() error {
+		daIncludedToSave := make(map[string]uint64)
+		c.daIncluded.Range(func(k, v any) bool {
+			keyStr, okKey := k.(string)
+			valUint64, okVal := v.(uint64)
+			if okKey && okVal {
+				daIncludedToSave[keyStr] = valUint64
+			}
+			return true
+		})
+		if err := saveMapGob(filepath.Join(folderPath, daIncludedFilename), daIncludedToSave); err != nil {
+			return fmt.Errorf("save %s: %w", daIncludedFilename, err)
 		}
-		return true
+		return nil
 	})
-	return saveMapGob(filepath.Join(folderPath, daIncludedFilename), daIncludedToSave)
+	return wg.Wait()
 }
 
 // LoadFromDisk loads the cache contents from disk from the specified folder.
 // It populates the current cache instance. If files are missing, corresponding parts of the cache will be empty.
 // It's the caller's responsibility to ensure that type T (and any types it contains)
 // are registered with the gob package if necessary (e.g., using gob.Register).
 func (c *Cache[T]) LoadFromDisk(folderPath string) error {
+	var wg errgroup.Group
 	// load items by height
-	itemsByHeightMap, err := loadMapGob[uint64, *T](filepath.Join(folderPath, itemsByHeightFilename))
-	if err != nil {
-		return fmt.Errorf("failed to load items by height: %w", err)
-	}
-	for k, v := range itemsByHeightMap {
-		c.itemsByHeight.Store(k, v)
-	}
-
+	wg.Go(func() error {
+		itemsByHeightMap, err := loadMapGob[uint64, *T](filepath.Join(folderPath, itemsByHeightFilename))
+		if err != nil {
+			return fmt.Errorf("failed to load %s : %w", itemsByHeightFilename, err)
+		}
+		for k, v := range itemsByHeightMap {
+			c.itemsByHeight.Store(k, v)
+		}
+		return nil
+	})
 	// load hashes
-	hashesMap, err := loadMapGob[string, bool](filepath.Join(folderPath, hashesFilename))
-	if err != nil {
-		return fmt.Errorf("failed to load hashes: %w", err)
-	}
-	for k, v := range hashesMap {
-		c.hashes.Store(k, v)
-	}
-
+	wg.Go(func() error {
+		hashesMap, err := loadMapGob[string, bool](filepath.Join(folderPath, hashesFilename))
+		if err != nil {
+			return fmt.Errorf("failed to load %s : %w", hashesFilename, err)
+		}
+		for k, v := range hashesMap {
+			c.hashes.Store(k, v)
+		}
+		return nil
+	})
 	// load daIncluded
-	daIncludedMap, err := loadMapGob[string, uint64](filepath.Join(folderPath, daIncludedFilename))
-	if err != nil {
-		return fmt.Errorf("failed to load daIncluded: %w", err)
-	}
-	for k, v := range daIncludedMap {
-		c.daIncluded.Store(k, v)
-		// Update max DA height during load
-		current := c.maxDAHeight.Load()
-		if v > current {
-			c.maxDAHeight.Store(v)
+	wg.Go(func() error {
+		daIncludedMap, err := loadMapGob[string, uint64](filepath.Join(folderPath, daIncludedFilename))
+		if err != nil {
+			return fmt.Errorf("failed to load %s : %w", daIncludedFilename, err)
 		}
-	}
-
-	return nil
+		for k, v := range daIncludedMap {
```

Comment:
are you going to do you suggestion here: https://github.com/evstack/ev-node/pull/2873#discussion_r2549091605. that would save us an extra PR :P